connectors/sources/github.py

```diff
@@ -0,0 +1,252 @@
+
+            for repo in public_repos:
+                try:
+                    await self.execute_getitem(
+                        request=ENDPOINTS["REPO"].format(repo_name=repo)
+                    )
+                except BadRequest:
+                    invalid_repo.append(repo)
+            return invalid_repo
+        except Exception as exception:
+            logger.exception(
+                f"Error while filtering repositories. Exception: {exception}."
+            )
+            raise
+
+    @cached_property
+    def _get_session(self):
+        self.session = aiohttp.ClientSession(headers=self.headers)
+        return GitHubAPI(
+            self.session,
+            requester="",
+            oauth_token=self.github_token,
+        )
+
+    async def ping(self):
+        await self.execute_getitem(request=ENDPOINTS["USER"])
+
+    async def close_session(self):
+        self._sleeps.cancel()
+        if self.session is None:
+            return
+        await self.session.close()
+        del self._get_session
+
+
+class GitHubDataSource(BaseDataSource):
+    """GitHub"""
+
+    name = "GitHub"
+    service_type = "github"
+
+    def __init__(self, configuration):
+        """Setup the connection to the GitHub instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self._sleeps = CancellableSleeps()
+        self.github_client = GitHubClient(configuration=configuration)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GitHub.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "github_token": {
+                "label": "GitHub Token",
+                "sensitive": True,
+                "order": 1,
+                "type": "str",
+                "value": "changeme",
+            },
+            "repositories": {
+                "display": "textarea",
+                "label": "List of repositories",
+                "order": 2,
+                "type": "list",
+                "value": "*",
+            },
+            "retry_count": {
+                "display_value": RETRIES,
+                "display": "numeric",
+                "label": "Maximum retries per request",
+                "order": 3,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": RETRIES,
+                "validations": [{"type": "less_than", "constraint": 10}],
+            },
+        }
+
+    async def _remote_validation(self):
+        """Validate scope of the configured Token and accessibility of repositories
+
+        Raises:
+            ConfigurableFieldValueError: Insufficient privileges error.
+        """
+        response_headers = await self.github_client.execute_request(
```

Comment:
I think it makes sense to have a high-level method here present in the client. For example: `get_user`

Reply:
Here we need `response_headers` only to validate the github token, and we are not fetching the user's data. So, WDYT, do we need to add the method `get_user`.

Reply:
But we want `response_headers` only and `execute_request` returns only `response_headers`. So, WDYT do we need to add `get_user` method.